src/Stratis.Bitcoin.Features.MemoryPool/MempoolFeature.cs

```diff
@@ -171,9 +171,9 @@ public static IFullNodeBuilder UseMempool(this IFullNodeBuilder fullNodeBuilder)
                         services.AddSingleton<BlockPolicyEstimator>();
                         services.AddSingleton<IMempoolValidator, MempoolValidator>();
                         services.AddSingleton<MempoolOrphans>();
-                        services.AddSingleton<MempoolManager>();
-                        services.AddSingleton<IPooledTransaction, MempoolManager>();
-                        services.AddSingleton<IPooledGetUnspentTransaction, MempoolManager>();
+                        services.AddSingleton<MempoolManager>()
+                            .AddSingleton<IPooledTransaction, MempoolManager>(provider => provider.GetService<MempoolManager>())
+                            .AddSingleton<IPooledGetUnspentTransaction, MempoolManager>(provider => provider.GetService<MempoolManager>());
```

Comment:
Is it correct that these ones are chained on to the previous addsingleton and the others here are on their own line of code?